extensions/pkg/controller/worker/genericactuator/interface.go

```diff
@@ -43,10 +43,17 @@ type WorkerDelegate interface {
 	// GenerateMachineDeployments generates the configuration for the desired machine deployments.
 	GenerateMachineDeployments(context.Context) (worker.MachineDeployments, error)
 
-	// GetMachineImages returns the list of used machine images for this `Worker` resource. It will be stored in the
-	// `.status.providerStatus` field of the `Worker` resource such that the controller can look up its provider-specific
-	// machine image information in case the required version has been removed from its componentconfig.
-	GetMachineImages(context.Context) (runtime.Object, error)
+	// UpdateMachineImagesStatus will store a list of used in machine images for
+	// this Worker resource in its provider status. The controller can look up its
+	// provider-specific machine image information in case the required version
+	// has been removed from its componentconfig.
```

Comment:
Nit

```suggestion
	// has been removed from the `CloudProfile`.
```

Since https://github.com/gardener-attic/gardener-extensions/pull/455 the machine images are not maintained in the componentconfig of the extension controller.